Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/manifoldness.h

```diff
@@ -386,8 +386,241 @@ OutputIterator non_manifold_vertices(const PolygonMesh& pm,
+    halfedge_descriptor ih = h, done = ih;
+    int border_counter = 0;
+    do
+    {
+      put(visited_halfedges, ih, true);
+      if(is_border(ih, pmesh))
+        ++border_counter;
+
+      ih = prev(opposite(ih, pmesh), pmesh);
+    }
+    while(ih != done);
+
+    if(border_counter > 1)
+      is_non_manifold = true;
+
+    if(is_non_manifold)
+    {
+      *out++ = h;
+      known_nm_points.insert(p);
+
+      if(verbose)
+        std::cout << "Non-manifold star centered at " << p << std::endl;
+    }
+  }
+
+  return out;
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////
+
+/// \ingroup PMP_repairing_grp
+///
+/// collects the non-manifold edges (if any) present in the mesh.
+///
+/// An edge is considered non-manifold if there exists another edge sharing the same geometry.
+/// Note that this means that two different edges with partial overlaps are thus not considered
+/// as non-manifold.
+///
+/// @tparam PolygonMesh a model of `HalfedgeListGraph`
+/// @tparam OutputIterator a model of `OutputIterator` holding objects of type
+///                         `boost::graph_traits<PolygonMesh>::%edge_descriptor`
+/// @tparam NamedParameters a sequence of \ref bgl_namedparameters "Named Parameters"
+///
+/// @param pmesh a polygon mesh
+/// @param out the output iterator that collects non-manifold edges
+/// @param np an optional sequence of \ref bgl_namedparameters "Named Parameters" among the ones listed below
+///
+/// \cgalNamedParamsBegin
+///   \cgalParamNBegin{vertex_point_map}
+///     \cgalParamDescription{a property map associating points to the vertices of `pm`}
+///     \cgalParamType{a class model of `ReadWritePropertyMap` with `boost::graph_traits<PolygonMesh>::%vertex_descriptor`
+///                    as key type and `%Point_3` as value type}
+///     \cgalParamDefault{`boost::get(CGAL::vertex_point, pm)`}
+///     \cgalParamExtra{If this parameter is omitted, an internal property map for `CGAL::vertex_point_t`
+///                     must be available in `PolygonMesh`.}
+///   \cgalParamNEnd
+///   \cgalParamNBegin{geom_traits}
+///     \cgalParamDescription{an instance of a geometric traits class}
+///     \cgalParamType{The traits class must provide the nested functor `Equal_3` to check
+///                    whether two points are identical and a function `Equal_3 equal_3_object()`.}
+///     \cgalParamDefault{a \cgal Kernel deduced from the point type, using `CGAL::Kernel_traits`}
+///     \cgalParamExtra{The geometric traits class must be compatible with the vertex point type.}
+///   \cgalParamNEnd
+/// \cgalNamedParamsEnd
+///
+/// \sa `non_manifold_vertices()`
+/// \sa `geometrically_non_manifold_vertices()`
+///
+/// \return the output iterator.
+template <typename PolygonMesh, typename OutputIterator,
+          typename NamedParameters = parameters::Default_named_parameters>
+OutputIterator geometrically_non_manifold_edges(const PolygonMesh& pmesh,
```

Comment:
You should check `Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/Non_manifold_feature_map.h`